ui/src/wizard/AzureWizard.ts

```diff
@@ -32,51 +31,49 @@ export class AzureWizard<T> implements types.AzureWizard<T>, IInternalAzureWizar
     }
 
     public get currentStep(): number {
-        return this._finishedPromptSteps.length + 1;
+        return this._finishedPromptSteps.filter(s => s.prompted).length + 1;
     }
 
     public get totalSteps(): number {
-        return this._finishedPromptSteps.length + this._promptSteps.filter(s => s.shouldPrompt(this._wizardContext)).length + 1;
+        return this._finishedPromptSteps.filter(s => s.prompted).length + this._promptSteps.filter(s => s.shouldPrompt(this._wizardContext)).length + 1;
     }
 
     public async prompt(actionContext: types.IActionContext): Promise<void> {
         // Insert Wizard UI into ext.ui.rootUserInput - to be used instead of vscode.window UI
-        let oldRootUserInput: IRootUserInput = vscode.window;
-        if (ext.ui instanceof AzureUserInput) {
-            oldRootUserInput = ext.ui.rootUserInput;
-            ext.ui.rootUserInput = new AzureWizardUserInput(this);
-        }
+        // tslint:disable-next-line: strict-boolean-expressions
+        const oldRootUserInput: IRootUserInput = ext.ui.rootUserInput || vscode.window;
+        ext.ui.rootUserInput = new AzureWizardUserInput(this);
 
         try {
             let step: AzureWizardPromptStep<T> | undefined = this._promptSteps.pop();
             while (step) {
+                step.init();
+
                 if (step.shouldPrompt(this._wizardContext)) {
                     actionContext.properties.lastStepAttempted = `prompt-${step.constructor.name}`;
                     step.propertiesBeforePrompt = Object.keys(this._wizardContext).filter(k => !isNullOrUndefined(this._wizardContext[k]));
 
                     try {
                         const subWizard: types.ISubWizardOptions<T> | void = await step.prompt(this._wizardContext);
-                        this._finishedPromptSteps.push(step);
+                        step.prompted = true;
                         if (subWizard) {
                             this.addSubWizard(step, subWizard);
                         }
                     } catch (err) {
                         if (err instanceof GoBackError) {
-                            this._promptSteps.push(step);
-                            step = this.goBack();
+                            step = this.goBack(step);
                             continue;
                         } else {
                             throw err;
                         }
                     }
                 }
 
+                this._finishedPromptSteps.push(step);
                 step = this._promptSteps.pop();
             }
         } finally {
```

Comment:
Why is this setting ext.ui.rootUserInput to vscode.window when it was undefined before?

Reply:
Realistically it didn't matter since `vscode.window` was already the default, but I agree the logic was weird. Fixed